Generated defaults fail to handle enums with negative int & large ulong

Repro `fbs` (w/ latest 7.1.0 compiler & runtime NuGet pkg):

```fbs
namespace Test;

enum IntEnum : int { Neg1 = -1 }
enum UlongEnum : ulong { Bit63Set = 0xCBF29CE484222645 }

table DummyObject {
  S32:IntEnum = Neg1;
  U64:UlongEnum = Bit63Set;
}
```

Generated output:
```cs
// !! FLATSHARP CODE GENERATION FAILED. THIS FILE MAY CONTAIN INCOMPLETE OR INACCURATE DATA !!
/* Error: 
FlatSharp.FlatSharpCompilationException: FlatSharp failed to generate proper C# for your schema.

FlatSharp compilation error: (82,53): error CS0119: 'IntEnum' is a type, which is not valid in the given context, Context = "Test.IntEnum"
FlatSharp compilation error: (82,52): error CS0075: To cast a negative value, you must enclose the value in parentheses., Context = "(Test.IntEnum) - 1"
FlatSharp compilation error: (82,53): error CS0119: 'IntEnum' is a type, which is not valid in the given context, Context = "Test.IntEnum"
FlatSharp compilation error: (35,20): error CS0031: Constant value '-3750763034362894779' cannot be converted to a 'ulong', Context = "-3750763034362894779"
FlatSharp compilation error: (55,25): error CS0119: 'IntEnum' is a type, which is not valid in the given context, Context = "Test.IntEnum"
FlatSharp compilation error: (55,24): error CS0075: To cast a negative value, you must enclose the value in parentheses., Context = "(Test.IntEnum) - 1"
FlatSharp compilation error: (55,25): error CS0119: 'IntEnum' is a type, which is not valid in the given context, Context = "Test.IntEnum"

   at FlatSharp.CodeGen.RoslynSerializerGenerator.ThrowOnEmitFailure(EmitResult result, SyntaxTree syntaxTree, Func`1 cSharpFactory) in D:\a\FlatSharp\FlatSharp\src\FlatSharp\Serialization\RoslynSerializerGenerator.cs:line 356
   at FlatSharp.CodeGen.RoslynSerializerGenerator.CompileAssembly(String sourceCode, Boolean enableAppDomainIntercept, Assembly[] additionalReferences) in D:\a\FlatSharp\FlatSharp\src\FlatSharp\Serialization\RoslynSerializerGenerator.cs:line 229
   at FlatSharp.Compiler.FlatSharpCompiler.<>c__DisplayClass17_1.<CreateCSharp>b__3() in D:\a\FlatSharp\FlatSharp\src\FlatSharp.Compiler\FlatSharpCompiler.cs:line 561
   at FlatSharp.Compiler.FlatSharpCompiler.Instrument[T](String step, CompilerOptions opts, Func`1 callback) in D:\a\FlatSharp\FlatSharp\src\FlatSharp.Compiler\FlatSharpCompiler.cs:line 647
   at FlatSharp.Compiler.FlatSharpCompiler.CreateCSharp(List`1 bfbs, String inputHash, CompilerOptions options) in D:\a\FlatSharp\FlatSharp\src\FlatSharp.Compiler\FlatSharpCompiler.cs:line 0
   at FlatSharp.Compiler.FlatSharpCompiler.RunCompiler(CompilerOptions options) in D:\a\FlatSharp\FlatSharp\src\FlatSharp.Compiler\FlatSharpCompiler.cs:line 118
*/
```

Expected behavior: Compiler would be able to generate its code with a negative default enum value, and high-ulong enum value. Same behavior exists for signed `byte` type.

Changing the default value to `0` generates the code as expected. Removing the enum and using the primitive directly (int/ulong) & default value works as expected too.

--- a/src/FlatSharp.Compiler/SchemaModel/EnumSchemaModel.cs
+++ b/src/FlatSharp.Compiler/SchemaModel/EnumSchemaModel.cs
@@ -25,6 +25,7 @@ public class EnumSchemaModel : BaseSchemaModel
     private readonly Dictionary<string, EnumVal> nameValueMap;
     private readonly string underlyingType;
     private readonly IEnumerable<string>? documentation;
+    private readonly FlatBufferEnum @enum;
 
     private EnumSchemaModel(Schema.Schema schema, FlatBufferEnum @enum) : base(schema, @enum.Name, new FlatSharpAttributes(@enum.Attributes))
     {
@@ -37,6 +38,7 @@ public class EnumSchemaModel : BaseSchemaModel
         this.documentation = @enum.Documentation;
         this.DeclaringFile = @enum.DeclarationFile;
         this.FriendlyName = @enum.OriginalName ?? @enum.Name;
+        this.@enum = @enum;
 
         this.AttributeValidator.ExternValidator = _ => AttributeValidationResult.Valid;
     }
@@ -79,7 +81,14 @@ public class EnumSchemaModel : BaseSchemaModel
                     writer.AppendSummaryComment(item.Value.Documentation);
                 }
 
-                writer.AppendLine($"{item.Key} = {item.Value.Value},");
+                if (this.@enum.UnderlyingType.BaseType == BaseType.ULong)
+                {
+                    writer.AppendLine($"{item.Key} = {(ulong)item.Value.Value},");
+                }
+                else
+                {
+                    writer.AppendLine($"{item.Key} = {item.Value.Value},");
+                }
             }
         }
     }

--- a/src/FlatSharp.Compiler/SchemaModel/PropertyFieldModel.cs
+++ b/src/FlatSharp.Compiler/SchemaModel/PropertyFieldModel.cs
@@ -206,7 +206,7 @@ public record PropertyFieldModel
                 return "true";
             }
 
-            return $"({typeName}){defaultInt}";
+            return $"({typeName})({defaultInt})";
         }
     }
 

--- a/src/Tests/FlatSharpEndToEndTests/DefaultValues/DefaultValues.fbs
+++ b/src/Tests/FlatSharpEndToEndTests/DefaultValues/DefaultValues.fbs
@@ -1,0 +1,26 @@
+﻿namespace FlatSharpEndToEndTests.DefaultValues;
+
+enum Int64Enum : long {
+   NegOne = -1,
+   LongMax = 9223372036854775807
+}
+
+enum UInt64Enum : ulong {
+   Zero = 0,
+   Max = 18446744073709551615
+}
+
+table Defaults
+{
+   EnumLongNegOne : Int64Enum = NegOne;
+   EnumLongMax : Int64Enum = LongMax;
+
+   EnumULongZero : UInt64Enum = Zero;
+   EnumULongMax : UInt64Enum = Max;
+
+   LongNegOne : long = -1;
+   LongMax : long = 9223372036854775807;
+
+   ULongZero : ulong = 0;
+   ULongMax : ulong = 18446744073709551615;
+}

--- a/src/Tests/FlatSharpEndToEndTests/DefaultValues/DefaultValuesTests.cs
+++ b/src/Tests/FlatSharpEndToEndTests/DefaultValues/DefaultValuesTests.cs
@@ -1,0 +1,38 @@
+﻿/*
+ * Copyright 2023 James Courtney
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace FlatSharpEndToEndTests.DefaultValues;
+
+public class DefaultValuesTests
+{
+    [Fact]
+    public void DefaultValue_Enum_Negative()
+    {
+        Defaults defaults = new();
+
+        Assert.Equal(long.MaxValue, (long)defaults.EnumLongMax);
+        Assert.Equal((long)-1, (long)defaults.EnumLongNegOne);
+
+        Assert.Equal(long.MaxValue, defaults.LongMax);
+        Assert.Equal(-1, defaults.LongNegOne);
+
+        Assert.Equal(0ul, defaults.ULongZero);
+        Assert.Equal(0ul, defaults.ULongMax);
+
+        Assert.Equal(0ul, (ulong)defaults.EnumULongZero);
+        Assert.Equal(0ul, (ulong)defaults.EnumULongMax);
+    }
+}

--- a/src/Tests/FlatSharpEndToEndTests/DefaultValues/DefaultValuesTests.cs
+++ b/src/Tests/FlatSharpEndToEndTests/DefaultValues/DefaultValuesTests.cs
@@ -30,9 +30,23 @@ public class DefaultValuesTests
         Assert.Equal(-1, defaults.LongNegOne);
 
         Assert.Equal(0ul, defaults.ULongZero);
+
+        // FlatC bug: ulong values over 2^63 are not supported.
         Assert.Equal(0ul, defaults.ULongMax);
 
         Assert.Equal(0ul, (ulong)defaults.EnumULongZero);
+
+        // FlatC bug: ulong values over 2^63 are not supported.
         Assert.Equal(0ul, (ulong)defaults.EnumULongMax);
+    }
+
+    [Fact]
+    public void Enum_Values()
+    {
+        Assert.Equal(-1L, (long)Int64Enum.NegOne);
+        Assert.Equal(long.MaxValue, (long)Int64Enum.LongMax);
+
+        Assert.Equal(0UL, (ulong)UInt64Enum.Zero);
+        Assert.Equal(ulong.MaxValue, (ulong)UInt64Enum.Max);
     }
 }

--- a/src/common.props
+++ b/src/common.props
@@ -13,8 +13,8 @@
   
   <PropertyGroup>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
-    <Version>7.1.0</Version>
-    <PackageVersion>7.1.0</PackageVersion>
+    <Version>7.1.1</Version>
+    <PackageVersion>7.1.1</PackageVersion>
     <AssemblyVersion>$(Version)</AssemblyVersion>
     <Authors>James Courtney</Authors>
     <Description>FlatSharp is a fast, idiomatic implementation of the FlatBuffer binary format.</Description>

